### ISSUE TITLE
Issue on FF45, Angular, Tooltips

### ISSUE THREAD

**Issue by FrankMerema** [amcharts/amcharts4#802]
For my current project we need to support FF 45. We are using Angular 7 and everything works super nice except tooltips on hovering over a column. This issue occurs only when you are using the Angular router and the chart is not on the first page.

Link below is minimal setup in combination with angular + router.
https://stackblitz.com/edit/gantt-chart-tooltip-bug

Open this link in FF45 and hover over a column to see the issue:
https://gantt-chart-tooltip-bug.stackblitz.io/dashboard

I have no clue what is going wrong so hopefully you guys can help me :)

**Comment by martynasma** [comment]
Two things jump out.

1) Your app uses older version of amCharts 4.

2) It does not have `ngOnDestroy` method. You should definitely have one:

```
  ngOnDestroy() {
    this.zone.runOutsideAngular(() => {
      if (this.chart) {
        this.chart.dispose();
      }
    });
  }
```

Bonus note: You should probably also run all the chart updating code inside `this.zone.runOutsideAngular(() => { ... })`

Let me know if any of that helps with your age old FF issue ;)

**Comment by FrankMerema** [comment]
I have updated all the things you mentioned, but still no success :( Maybe any other tips?

**Comment by martynasma** [comment]
Bummer. I was hoping this would do the trick :)

This is not happening with current version of FF, and even with FF ESR.

It's only happening with the version that is, like, three years old.

And tooltips show up, they just don not look good. I did test it. So it's not like piece of information is missing.

We are still going to look into it, but I'm afraid we can't treat it as a priority :/

**Comment by FrankMerema** [comment]
I completely understand, I have found an other bug on FF 45 which I will investigate and report. Hope you guys will find a solution :)

Thanks for the good help again!

**Comment by jimmykane** [comment]
Same happens on safari Latest + ALL IOS devices (FF,Chrome + MOZ) 

![screenshot 2019-01-20 at 19 01 00](https://user-images.githubusercontent.com/1268723/51443213-c6ce2d00-1ce5-11e9-9209-311cc2d281d6.png)

**Comment by jimmykane** [comment]
@FrankMerema  if you would use multiple yAxis do they also leak in Safari plus AG7? 

see https://github.com/amcharts/amcharts4/issues/841

I somehow tend to relate those issues, as FF 45 is kinda the same webit (if not mistaken) as safari etc. OR at least also has the same issue exactly with the tooltips

**Comment by FrankMerema** [comment]
@martynasma any updates?

**Comment by martynasma** [comment]
Seems related to #841 and #822.

**Comment by FrankMerema** [comment]
Tested it and I confirm that it's fixed in 4.1.5!

**Comment by martynasma** [comment]
Folks, just released 4.1.5 which should fix the issues with Angular and Safari/FF45.

At least it's so in our tests.

I'm gonna go ahead and close this, but feel free to reopen if you're still seeing issues after upgrading.
